File: CreditCardValidator.java
import java.util.Scanner;
import static java.lang.Long.parseLong;
import static java.lang.System.out;
import static java.text.MessageFormat.format;

public class CreditCardValidator {
	public static void main(String[] args) {
		
		Scanner input = new Scanner(System.in);
		out.print("\nEnter a credit card number without spaces or dashes: ");
		var creditCardNumber = input.nextLine();
		input.close();
		
		var oddDigitsSum = 0;
		var evenDigitsSum = 0;
		long CreditCardNumberLong = parseLong(creditCardNumber);
		
		for (var i = 0; i < creditCardNumber.length(); i++) {
			if (i % 2 == 0) oddDigitsSum += CreditCardNumberLong % 10;
			else if (CreditCardNumberLong % 10 * 2 > 9)
				evenDigitsSum += (CreditCardNumberLong % 10 * 2) % 10 + (CreditCardNumberLong % 10 * 2) / 10;
			else evenDigitsSum += CreditCardNumberLong % 10 * 2;
			CreditCardNumberLong /= 10;
		}
		
		var totalSum = oddDigitsSum + evenDigitsSum;
		out.println(format("\nThe credit card number: {0}", creditCardNumber));
		if (totalSum % 10 == 0) out.println("The card is valid\n");
		else out.println("The credit card is INVALID\n");
	}
}

File: CreditCardValidator.py
odd_digits_sum: int = 0
even_digits_sum: int = 0
total_sum: int = 0

credit_card_number: str = input(
    "Enter a credit card number without spaces or dashes: ")
credit_card_number = credit_card_number[::-1]

for x in credit_card_number[::2]:
    odd_digits_sum += int(x)

for x in credit_card_number[1::2]:
    x = int(x) * 2
    if x >= 10:
        even_digits_sum += (1 + (x % 10))
    else:
        even_digits_sum += x

total_sum = odd_digits_sum + even_digits_sum

print(f'\nThe credit card number: {credit_card_number[::-1]} is valid\n') if total_sum % 10 == 0 else print(
    f'The credit card number: {credit_card_number[::-1]} is invalid\n')
